src/cli/compare_fixtures.py

```diff
@@ -70,19 +78,16 @@ def remove_fixture_from_file(file: Path, test_case_id: str):
         raise KeyError(f"Test case {test_case_id} not found in {file}") from None
 
 
-def remove_fixture(
-    folder: Path,
-    index: IndexFile,
-    fixture_hash: HexNumber,
-    dry_run: bool,
-):
-    """Remove a single fixture from a folder that matches the given hash."""
-    test_case = pop_by_hash(index, fixture_hash)
-    test_case_file = folder / test_case.json_path
-    if dry_run:
-        print(f"Remove {test_case.id} from {test_case_file}")
-    else:
-        remove_fixture_from_file(test_case_file, test_case.id)
+def batch_remove_fixtures_from_files(removals_by_file):
+    """Batch process file removals to minimize I/O."""
+    for file_path, test_case_ids in removals_by_file.items():
+        try:
+            full_file = json.loads(file_path.read_text())
+            for test_case_id in test_case_ids:
+                full_file.pop(test_case_id, None)
+            file_path.write_text(json.dumps(full_file, indent=2))
```

Comment:
```suggestion
            if len(full_file) > 0:
                file_path.write_text(json.dumps(full_file, indent=2))
            else:
                file_path.unlink()
```
Just to clean up the empty files at the end.